src/workers/notifications/squadPostModerationRejected.ts

```diff
@@ -0,0 +1,31 @@
+import { generateTypedNotificationWorker } from './worker';
+import { NotificationType } from '../../notifications/common';
+import { NotificationPostModerationContext } from '../../notifications';
+import { User } from '../../entity';
+import { SquadPostModerationStatus } from '../../entity/SquadPostModeration';
+
+const worker =
+  generateTypedNotificationWorker<'api.v1.squad-post-moderation-submitted'>({
+    subscription: 'api.v1.squad-post-moderation-rejected-notification',
+    handler: async ({ post }, con) => {
+      if (
+        !post?.createdById ||
+        post.status !== SquadPostModerationStatus.Rejected
```

Comment:
Don't we need this check for the approved one as well? (to say it's not approved yet?)
Also maybe split it like the above one, i like how readable it is.

Reply:
Right, let me check any missing checks.